### DISCUSSION TITLE
Can I still use dynamic completion?

### DISCUSSION THREAD

**Discussion by alexreg** [scop/bash-completion#865] [Q&A]
Is there any way to use dynamic completion (`complete -D -F ...`) in `.bash_completion` or `.bashrc` whilst simultaneously maintaining auto-loading of completion scripts from the `.local/share/bash-completion/completions` dir? Trying that now, it seems that bash-completion doesn't do its usual job of using the completion scripts in that standard dir. For example, adding the following code to `~/.bash_completion` breaks the regular functionality.
```bash
_extra_completion_loader()
{
    return 124 # same result returning 0 or 1
}
complete -D -F _extra_completion_loader
```

**Comment by alexreg** [comment]
It seems perhaps that `complete -D` can only be used once – or rather, subsequent calls override previous ones. That's unfortunate if so (not very extensible), but is something like the following recommended in that case?
```bash
_new_completion_loader()
{
    # custom logic here, with possible early return

    # fall back to bash-completion loader
    _completion_loader "$@"
}
complete -D -F _new_completion_loader
```

**Reply by scop** [reply]
I don't have particular recommendations for that, but I can see how your example could work. (The "fall back" should probably read `_completion_loader "$@"` though.)

**Reply by alexreg** [reply]
Yep, that's what I meant indeed. Anyway, thanks for your reply.

**Reply by akinomyoga** [reply]
> Yep, that's what I meant indeed.

I edited the original reply.
